nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -166,6 +177,21 @@ def setup_optimizer_param_groups(self):
         """ModelPT override. Optimizer will get self._optimizer_param_groups"""
         self._optimizer_param_groups = get_params_for_weight_decay_optimization([self.model])
 
+    def setup_optimization(
+            self,
+            optim_config: Optional[Union[DictConfig, Dict]] = None,
+            optim_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        optim_kwargs = {} if optim_kwargs is None else optim_kwargs.copy()
+        if self.with_distributed_adam:
+            optim_kwargs['process_group'] = parallel_state.get_data_parallel_group()
```

Comment:
This is being called after data parallel groups have been created right?

Reply:
The model parallel groups must have been initialized before constructing the model, and [I believe the data parallel groups are created at the same time](https://github.com/NVIDIA/apex/blob/a4f42586d3b4252dc1c8d89bbc693307139959ec/apex/transformer/parallel_state.py#L185).